### PR TITLE
[ONNX] Update ONNX documentation

### PR DESCRIPTION
**Pull request by KsenijaS**
Add information about .data field

### CODE REVIEW

**Review by BowenBao — COMMENTED** [review]
looks good, thanks. Could you maybe create sub section titles for avoiding numpy, and avoiding .data?

File: torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp [hunk 1]
```diff
@@ -492,7 +492,25 @@ static void PrepareForRemoveMutations(MutationRemover& mr, Block* b) {
                   << (*it)->debugName() << "'. This changes graph semantics."
                   << std::endl;
 
-        Node* newNode = addDummyCloneToBlock(b, input);
+        Node* newNode = nullptr;
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp:495]: why are these included, rebase issue?

**Review by BowenBao — APPROVED** [review]

File: docs/source/onnx.rst [hunk 2]
```diff
@@ -274,6 +277,13 @@ In addition, Dropout layer need defined in init function so that inferencing can
         def forward(self, x):
             x = self.dropout(x)
 
+Avoid using .data field
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The .data field is an old field that is kept for backward compatibility and should be avoided when writing models.
+It's usage is dangerous and can make computations wrong, furthermore it can produce an incorrect trace graph and
```
Comment (jiafatom) [inline docs/source/onnx.rst:284]: nit: It's -> Its

Comment (BowenBao) [inline docs/source/onnx.rst:284]:
```suggestion
Its usage is dangerous and can make computations wrong, furthermore it can produce an incorrect trace graph and
```

**Review by jiafatom — APPROVED** [review]
